### ISSUE TITLE
S3168 FP: Method used as TimerCallback not recognized

### ISSUE THREAD

**Issue by csaba-sagi-sonarsource** [SonarSource/sonar-dotnet#5432]
### Description

Rule S3168 does not recognize the async void method as the `TimerCallback`.

### Repro steps

```
public void SomeMethod()
{
    var _timer = new System.Threading.Timer(RunOnceAsync);
}

private async void RunOnceAsync(object? _)
{
}
```

### Expected behavior

No issue is raised.

### Actual behavior

An issue is raised.

### Known workarounds

None

### Related information

* C#/VB.NET Plugins version: 8.36.0.43782
